src/common/config.cc

```diff
@@ -874,7 +874,7 @@ int md_config_t::set_val(const char *key, const char *val, bool meta)
   }
 
   // couldn't find a configuration option with key 'key'
-  return -ENOENT;
+  return -EINVAL;
```

Comment:
@liupan1111 I am not sure @tchaikov meant this, but instead of changing `ENOENT` to `EINVAL` in `md_config_t::set_val` you could change the error messages generated by the caller. In this case this is `CephContext::do_command` from `common/ceph_context.cc`. I.e. instead of:
```
       if (r < 0) {
          f->dump_stream("error") << "error setting '" << var << "' to '" << valstr << "': " << cpp_strerror(r);
```
make something like below:
```
       if (r == -ENOENT) {
          f->dump_stream("error") << "'" << var << "' is  unknown";
       } else if (r < 0) {
          f->dump_stream("error") << "error setting '" << var << "' to '" << valstr << "': " << cpp_strerror(r);
```
Just changing it to `EINVAL` does not make things better. When I see `"error setting 'rbd_test' to '1':  (22) Invalid argument"`, I think that value '1' is invalid, not the parameter name.